### PR TITLE
IndexScrollbar: fix for scroll to select index

### PR DESCRIPTION
**Pull request by TomaszLukawskiSam**
[Issue] N/A
[Problem] (UI Component) IndexScrollBar display wrong index
[Solution] There was not implementation of indexScrollbar with ArcListview.
The implementation has been added.

Signed-off-by: Tomasz Lukawski <t.lukawski@samsung.com>

### CODE REVIEW

**Comment by TomaszLukawskiSam** [comment]
The patch has still one defect: "drag" event is not triggering when popup with letter is visible.

**Comment by TomaszLukawskiSam** [comment]
> The patch has still one defect: "drag" event is not triggering when popup with letter is visible.

Defect was fixed.

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js [hunk 1]
```diff
@@ -46,17 +47,43 @@
...
+				}
+			} else {
+				index = listview.findItemIndexByDivider(divider);
+				if (index > -1) {
+					listview.scrollToPosition(index);
+				}
+			}
+		});
+
+		indexScrollbarElement.addEventListener("show", function () {
```
Comment (grzegorz-czajkowski) [inline examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:73]: Are those "show" and "hide" events are needed to be handled to solve this issue? Wondering if they shouldn't be part of different commit/PR if possible.

File: src/js/profile/wearable/widget/wearable/ArcListview.js [hunk 2]
```diff
@@ -722,6 +722,20 @@
...
+			 * @method findItemIndexByDivider
+			 * @param {HTMLElement} dividerElement
+			 * @memberof ns.widget.wearable.ArcListview
+			 */
+			prototype.findItemIndexByDivider = function (dividerElement) {
+				var result = this._state.separators.filter(function (item) {
+					return item.itemElement.element === dividerElement;
+				});
+
+				return (result.length) ? result[0].insertBefore : -1;
```
Comment (grzegorz-czajkowski) [inline src/js/profile/wearable/widget/wearable/ArcListview.js]: No brackets needed.

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js [hunk 3]
```diff
@@ -46,17 +47,43 @@
...
 		}
 
+		listview = tau.engine.getBinding(listviewElement);
+
 		// Add IndexScrollbar index "select" event handler.
 		indexScrollbarElement.addEventListener("select", function (ev) {
 			var divider,
-				idx = ev.detail.index;
+				idx = ev.detail.index.trim(),
+				index;
```
Comment (grzegorz-czajkowski) [inline examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:56]: index = -1 ?

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js [hunk 4]
```diff
@@ -5,7 +5,8 @@
 		listviewElement = document.getElementById("list1"),
 		isCircle = tau.support.shape.circle,
 		scroller,
-		indexScrollbar;
+		indexScrollbar,
+		listview;
```
Comment (grzegorz-czajkowski) [inline examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:9]: Can we initialize it here?
 listview = tau.engine.getBinding(listviewElement);

**Review by grzegorz-czajkowski — COMMENTED** [review]
Added comments for consideration.

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js (hunk 4 above)
Comment (TomaszLukawskiSam) [reply examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:9]: We cannot initialize anything in this way before "pageshow" event.

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js (hunk 3 above)
Comment (TomaszLukawskiSam) [reply examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:56]: We don't have to initialize this value twice.

File: examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js (hunk 1 above)
Comment (TomaszLukawskiSam) [reply examples/wearable/UIComponents/contents/navcontrols/indexscrollbar/indexScrollbar.js:73]: This is base for this change.

File: src/js/profile/wearable/widget/wearable/ArcListview.js (hunk 2 above)
Comment (TomaszLukawskiSam) [reply src/js/profile/wearable/widget/wearable/ArcListview.js]: Ok. I'll change this.

**Review by grzegorz-czajkowski — APPROVED** [review]
Looks fine.

**Comment by lslachciak** [comment]
@TomaszLukawskiSam as we discussed it would be good to have index reacting not only on quick rotate events but on quick swipe also

**Comment by TomaszLukawskiSam** [comment]
> @TomaszLukawskiSam as we discussed it would be good to have index reacting not only on quick rotate events but on quick swipe also

New changes have been added. Could I ask you for review again.

**Review by lslachciak — APPROVED** [review]
works well

File: src/js/profile/wearable/widget/wearable/CircularIndexScrollbar.js [hunk 5]
```diff
@@ -145,7 +145,17 @@
 				engine = ns.engine,
 				utilsEvents = ns.event,
 				eventTrigger = utilsEvents.trigger,
+				gesture = ns.event.gesture,
 				prototype = new BaseWidget(),
+				DRAG_STEP_TO_VALUE = 60,
+				// FAST_SWIPE_THRESHOLD = 4,
```
Comment (TomaszLukawskiSam) [inline src/js/profile/wearable/widget/wearable/CircularIndexScrollbar.js]: Remove this comment.

**Review by lslachciak — APPROVED** [review]
